[Nigeltzy] iP

Nigeltzy opened the pull request:
> Greetings loved ones. Let's take a journey.

# Welcome to the ShnoopBot

## This is where a description would normally go.

### And this is where even more details may be given.

#### Even more details.

##### More.

###### The end.

``` Java
String s = "Below this code block are some ways you can read this description";
System.out.println(s);
return "Goodbye";
```

1. `Read` it with a **grain** of salt. :salt:
2. `Read` it while *checking* the *check* boxes to ensure it contains the [necessary GFMD elements](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html). 📓 
3. ~Not `read` it~. 😢 

Why you should read it:
- Why not.
- No particular reason.
- Adds information to description

- [x] Heading
- [x] Bullet List
- [x] Numbered List
- [x] Fenced Code Block (with syntax highlighting)
- [x] Task List
- [x] Emoji
- [x] Blockquote
- [x] Hyperlink
- [x] Inline Code
- [x] Some text formatting: bold, italic, strikethrough etc.


> [!NOTE]
> This bot has no affiliation with Snoop Dog or the song California Girls.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Nigeltzy committed:
Update Duke.java to Shnoop.java

Nigeltzy committed:
Update Shnoop.java with replies

Nigeltzy committed:
Shnoop.java: Add todo functionality

Nigeltzy committed:
Add Tasks functionality

Nigeltzy committed:
Add supporting classes, Simplify Shnoop.java

Nigeltzy committed:
Add multi-type task functionality, simplify & extract Shnoop logic flow

Nigeltzy committed:
Attempt SUT

Nigeltzy committed:
Add exceptions handling

Nigeltzy committed:
Add remove functionality

Nigeltzy committed:
Add enums

Nigeltzy committed:
Add file saving functionality

Nigeltzy committed:
Merge branch 'branch-Level-7'

Nigeltzy committed:
Add DateTime functionality

Nigeltzy committed:
Merge branch 'branch-Level-8'

Nigeltzy committed:
Remove redundant function

Nigeltzy committed:
Extract clases (midway)

Nigeltzy committed:
Remove line

ty4g1 commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,25 @@
+public class Todo extends Task {
+    public Todo (String description) throws EmptyDescriptionException {
```
There's an extra whitespace after `Todo` here, should be `public Todo(String description)`. Similar issue in the other overloaded constructor.

ty4g1 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,92 @@
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
```
Missing JavaDocs header for this class. Seems to be the case for all other classes as well. I'd recommend adding header JavaDocs comments for each class to comply with the code quality requirements.

ty4g1 reviewed (commented):
LGTM, just a few nits with JavaDocs and style. Another thing I'd recommend is to have a `Ui` class to store all the strings/messages that the chatbot prints out. 
Since if any changes need to be made in the future to these messages, they can easily be found, without having to comb through the code.

Nicholascyx commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+public class Deadline extends Task {
+    protected String by;
+    public Deadline (String description, String by) throws EmptyDescriptionException {
```
There is a white space between Deadline and (String description, String by), it should be Deadline(String description, String by).

Nicholascyx replied:
There is a white space between Deadline and (String description, String by), it should be Deadline(String description, String by).

Nicholascyx commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,8 @@
+
+ ✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿ 
+ ✿ It's Shnoop, my dawg. I'm all up on ya. Whatchu need? ✿ 
+ ✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿-✿
```
Nice art for the separating lines!

Nicholascyx commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    @Override
+    public String toUString() {
```
toUString might be a bit hard to understand? Is it toUserString()?

Nigeltzy committed:
Refactor classes

Nigeltzy committed:
Merge branch 'branch-A-MoreOOP'

Nigeltzy committed:
Refactor into different packages

Nigeltzy committed:
Merge branch 'add-gradle-support'

Nigeltzy committed:
Add proper JavaDocs for all relevant methods.

Nigeltzy committed:
Merge branch 'add-gradle-support'

Nigeltzy committed:
Merge branch 'master' of https://github.com/Nigeltzy/ip

Nigeltzy committed:
Add Gradle support

Nigeltzy committed:
Add simple Todo JUnit test

Nigeltzy committed:
Merge branch 'master' of https://github.com/Nigeltzy/ip

Nigeltzy committed:
Merge branch 'master' of https://github.com/Nigeltzy/ip

Nigeltzy committed:
Merge branch 'master' of https://github.com/Nigeltzy/ip

Nigeltzy committed:
Add equals() functionality, JUnit test

Nigeltzy committed:
Add JUnit testing

Nigeltzy committed:
Merge branch 'master' of https://github.com/Nigeltzy/ip

Nigeltzy committed:
Configured Jar creation

Nigeltzy committed:
Add FindCommand functionality

Nigeltzy committed:
Merge branch 'branch-Level-9'

Nigeltzy committed:
Update Parser.java

Nigeltzy committed:
Update JavaDocs

Nigeltzy committed:
Merge branch 'branch-A-JavaDoc'

Nigeltzy committed:
Merge branch 'branch-A-CodingStandard'

Nigeltzy committed:
Rectify all CheckStyle violations

Nigeltzy committed:
Merge branch 'branch-A-CheckStyle'

Nigeltzy committed:
Add JavaFX (Part 2)

Nigeltzy committed:
Add GUI functionality

Nigeltzy committed:
Merge branch 'branch-Level-10'

Nigeltzy committed:
Integrate assert testing within Shnoop.java

There are several classes pivotal to the functioning of the ChatBot.
If any of the classes do not function as intended, debugging would
be a difficult and complicated process.

Performing assumption checks at the start of the code would be most
useful.

As a step towards fool-proofing the application, let's perform assert
tests on one of the main classes that is invoked and initialised
from the very beginning. Doing so will check to see if any assumptions
do not hold true at the very base and beginning invocation of the
program.

Nigeltzy committed:
Refactor Parser.java method

There are several repeated / duplicated code within the method.

Refactoring the duplicated code improves readability of the method
and the class.

As a step towards improving readability, let's refactor such duplicated
code and lean towards method usage.

Nigeltzy committed:
Merge tag 'branch-A-Assertions'

Nigeltzy committed:
Merge branch 'master' into branch-A-CodeQuality

Nigeltzy committed:
Merge branch 'branch-A-CodeQuality'

Nigeltzy committed:
Add Sorting functionality for List Command

List command did not support sorted listing.
Users may want to view tasks in alphabetical order.

For a small starting point, let's add a basic sorting functionality
that allows users to input "1" after list to sort their tasks in
a simple defined alphabetical order. In future commits, we can improve
this sorting functionality to allow for more flexibility.

Nigeltzy committed:
Merge branch 'branch-C-Sort'

Nigeltzy committed:
Fix initial invalid command UI display

Nigeltzy committed:
Refactor Parser.java

Nigeltzy committed:
Parser.java: Add Javadocs

Nigeltzy committed:
Merge pull request #2 from Nigeltzy/branch-Merging-PR

Parser.java: Add Javadocs

Nigeltzy committed:
Change profile images

Nigeltzy committed:
Merge branch 'master' of https://github.com/Nigeltzy/ip

Nigeltzy committed:
Merge branch 'branch-Merging-PR'

Nigeltzy committed:
Configure website

Nigeltzy committed:
Merge branch 'master' of https://github.com/Nigeltzy/ip

Nigeltzy committed:
Configure README.md

Nigeltzy committed:
Add more JavaDocs comments

Nigeltzy committed:
Update README.md

Nigeltzy committed:
Fix README.md bug

Nigeltzy committed:
Acknowledge code author

Nigeltzy committed:
Update README.md

Nigeltzy committed:
Merge branch 'master' of https://github.com/Nigeltzy/ip

Nigeltzy committed:
Test README.md

Nigeltzy committed:
Update README.md
